test/functional/services/combo_box.js

```diff
@@ -0,0 +1,130 @@
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export function ComboBoxProvider({ getService }) {
+  const testSubjects = getService('testSubjects');
+  const find = getService('find');
+  const log = getService('log');
+  const retry = getService('retry');
+
+  // wrapper around EuiComboBox interactions
+  class ComboBox {
+
+    async set(comboBoxSelector, value) {
+      log.debug(`comboBox.set, comboBoxSelector: ${comboBoxSelector}`);
+      const comboBox = await testSubjects.find(comboBoxSelector);
+      await this.setElement(comboBox, value);
+    }
+
+    async setElement(comboBoxElement, value) {
+      log.debug(`comboBox.setElement, value: ${value}`);
+      await this._filterOptionsList(comboBoxElement, value);
+      await find.clickByCssSelector('.euiComboBoxOption');
+      await this.closeOptionsList(comboBoxElement);
+    }
+
+    async filterOptionsList(comboBoxSelector, filterValue) {
+      log.debug(`comboBox.filterOptionsList, comboBoxSelector: ${comboBoxSelector}, filter: ${filterValue}`);
+      const comboBox = await testSubjects.find(comboBoxSelector);
+      await this._filterOptionsList(comboBox, filterValue);
+      await this.closeOptionsList(comboBox);
+    }
+
+    async _filterOptionsList(comboBoxElement, filterValue) {
+      const input = await comboBoxElement.findByTagName('input');
+      await input.clearValue();
+      await input.type(filterValue);
+      await this._waitForOptionsListLoading(comboBoxElement);
+    }
+
+    async _waitForOptionsListLoading(comboBoxElement) {
+      await comboBoxElement.waitForDeletedByClassName('euiLoadingSpinner');
+    }
+
+    async getOptionsList(comboBoxSelector) {
+      log.debug(`comboBox.getOptionsList, comboBoxSelector: ${comboBoxSelector}`);
+      const comboBox = await testSubjects.find(comboBoxSelector);
+      await testSubjects.click(comboBoxSelector);
+      await this._waitForOptionsListLoading(comboBox);
+      const menu = await retry.try(
+        async () => await testSubjects.find('comboBoxOptionsList'));
+      const optionsText = await menu.getVisibleText();
+      await this.closeOptionsList(comboBox);
+      return optionsText;
+    }
+
+    async doesComboBoxHaveSelectedOptions(comboBoxSelector) {
+      log.debug(`comboBox.doesComboBoxHaveSelectedOptions, comboBoxSelector: ${comboBoxSelector}`);
+      const comboBox = await testSubjects.find(comboBoxSelector);
```

Comment:
Should this method also wait to make sure the list is fully loaded?  I hate to scatter that check everywhere but I don't know all the usages of the comboBox.  If it's getting populated based on results from an Elasticsearch query, it could take a second or so (especially on Cloud).